src/js/plugins/plugin.search.view.js

```diff
@@ -0,0 +1,328 @@
+class SearchView {
+  constructor(params) {
+    if (!params.selector) {
+      console.warn('BookReader::Search - SearchView must be passed a valid CSS selector');
+      return;
+    }
+
+    this.br = params.br;
+
+    // Search results are returned as a text blob with the hits wrapped in
+    // triple mustaches. Hits occasionally include text beyond the search
+    // term, so everything within the staches is captured and wrapped.
+    this.matcher = new RegExp('{{{(.+?)}}}', 'g');
+    this.cacheDOMElements();
+    this.bindEvents();
+
+    if (this.br.options.initialSearchTerm) {
+      this.setQuery(params.query);
+    }
+  }
+
+  cacheDOMElements() {
+    this.dom = {};
+
+    // The parent search tray in mobile menu
+    this.dom.searchTray: this.renderSearchTray(params.selector);
+    // Container for rendered search results
+    this.dom.results = this.dom.searchTray.querySelector('[data-id="results"]');
+    // Element used to display number of results
+    this.dom.resultsCount = this.dom.searchTray.querySelector('[data-id="results_count"]');
+    // Search input within the mobile search tray
+    this.dom.searchField = this.dom.searchTray.querySelector('[name="query"]');
+    // Waiting indicator displayed while waiting for a search request
+    this.dom.searchPending = this.dom.searchTray.querySelector('[data-id="searchPending"]');
+    // The element added to the mobile menu that is animated into view when
+    // the "search" nav item is clicked
+    this.dom.mobileSearch = this.buildMobileDrawer();
+    // Search input within the top toolbar. Will be removed once the mobile menu is replaced.
+    this.dom.toolbarSearch = this.buildToolbarSearch();
+  }
+
+  toggleSearchTray(bool) {
+    const state = typeof bool !== 'boolean' ? this.dom.searchTray.classList.contains('hidden') : bool;
+    this.dom.searchTray.classList.toggle('hidden', !state);
+  }
+
+  toggleResultsCount(bool) {
+    this.dom.resultsCount.classList.toggle('visible', bool);
+  }
+
+  updateResultsCount(results) {
+    this.dom.resultsCount.innerText = `(${results} result${results != 1 ? 's' : ''})`;
+    this.toggleResultsCount(true);
+  }
+
+  setQuery(query) {
+    this.dom.searchField.value = query;
+  }
+
+  emptyMatches() {
+    this.dom.results.innerHTML = '';
+  }
+
+  removeResultPins() {
+    this.br.$('.BRnavpos .BRsearch').remove();
+  }
+
+  clearSearchFieldAndResults() {
+    this.toggleResultsCount(false);
+    this.removeResultPins();
+    this.emptyMatches();
+    this.setQuery('');
+  }
+
+  renderSearchTray(selector) {
```

Comment:
Doc string for what this selector is used for; it looks like it gets converted to an ID?